unified-runtime/source/adapters/opencl/platform.cpp

```diff
@@ -86,38 +69,83 @@ UR_APIEXPORT ur_result_t UR_APICALL
 urPlatformGet(ur_adapter_handle_t *, uint32_t, uint32_t NumEntries,
               ur_platform_handle_t *phPlatforms, uint32_t *pNumPlatforms) {
 
-  cl_int Result =
-      clGetPlatformIDs(cl_adapter::cast<cl_uint>(NumEntries),
-                       cl_adapter::cast<cl_platform_id *>(phPlatforms),
-                       cl_adapter::cast<cl_uint *>(pNumPlatforms));
-
-  /* Absorb the CL_PLATFORM_NOT_FOUND_KHR and just return 0 in num_platforms */
-  if (Result == CL_PLATFORM_NOT_FOUND_KHR) {
-    Result = CL_SUCCESS;
-    if (pNumPlatforms) {
-      *pNumPlatforms = 0;
+  ur_adapter_handle_t Adapter = nullptr;
+  UR_RETURN_ON_FAILURE(urAdapterGet(1, &Adapter, nullptr));
+  if (Adapter && !(Adapter->NumPlatforms)) {
```

Comment:
we do specify PlatformGet as being thread safe so we should probably have this under mutex + a call_once

Reply:
Added a mutex guarding it.